pkg/runtime/scheduler/scheduler.go

```diff
@@ -254,6 +254,11 @@ func (s *Scheduler) connSchedulerHosts(ctx context.Context) (schedulerv1pb.Sched
 
 	resp, err := stream.Recv()
 	if err != nil {
+		if status.Code(err) == codes.Unimplemented {
+			// Ignore unimplemented error code as we are talking to an old server.
+			// TODO: @joshvanl: remove special case in v1.16.
+			return nil, s.addresses, nil
+		}
```

Comment:
I don't know how we were getting this error here in one of the tests, I would have expected it to occur on line 244 above. Some server config maybe?